test/python/test_tools_postcodes.py

```diff
@@ -58,15 +63,17 @@ def postcode_table(temp_db_conn, placex_table, word_table):
     return MockPostcodeTable(temp_db_conn)
 
 
-def test_postcodes_empty(dsn, postcode_table, tmp_path, tokenizer):
+def test_postcodes_empty(dsn, postcode_table, place_table,
+                         tmp_path, tokenizer):
     postcodes.update_postcodes(dsn, tmp_path, tokenizer)
 
     assert not postcode_table.row_set
 
 
-def test_postcodes_add_new(dsn, placex_table, postcode_table, tmp_path, tokenizer):
-    placex_table.add(country='xx', geom='POINT(10 12)',
-                     address=dict(postcode='9486'))
+def test_postcodes_add_new(dsn, postcode_table, placex_table, place_row,
+                           tmp_path, tokenizer):
+    placex_table.add(country='xx', geom='POINT(10 12)')
+    place_row(geom='SRID=4326;POINT(10 12)', address=dict(postcode='9486'))
```

Comment:
I might have missed it but do you also test the case somewhere where placex does not a corresponding line?

Also, consider adding a [pytest fixture with a factory](https://docs.pytest.org/en/stable/fixture.html#factories-as-fixtures)  for adding rows with postcodes, i.e the fixture would that wraps these two lines.

Reply:
I will add one test specially to test the call to get_country_code.